massa-execution-worker/src/execution.rs

```diff
@@ -1354,4 +1406,95 @@ impl ExecutionState {
     pub fn get_address_future_deferred_credits(&self, address: &Address) -> BTreeMap<Slot, Amount> {
         context_guard!(self).get_address_future_deferred_credits(address, self.config.thread_count)
     }
+
+    /// Get the execution statuses of both speculative and final executions
+    ///
+    /// # Return
+    ///
+    /// * A tuple of hashmaps with:
+    /// * first the statuses for speculative executions
+    /// * second the statuses for final executions
+    pub fn get_op_exec_status(&self) -> (HashMap<OperationId, bool>, HashMap<OperationId, bool>) {
```

Comment:
This was not done properly. Cf one of my previous comments.
In particular, the final state is not used as fallback for the active state (see how it's done in other functions here)

Reply:
this needs a followup issue

Reply:
#3707

Reply:
thx